### PR TITLE
[CI] Fix unit tests duplication generated by the Codecov workaround

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR fixes Codecov workaround, added in #21954.

### Motivation

#21954 duplicated unit tests when retries occured. 

#### What happened

**Context**: `gotestsum` allows us to re-run tests that failed during the run. But because of [this issue](https://github.com/gotestyourself/gotestsum/issues/274) it's rewriting the coverage files on re-run. The final coverage file was not accurate, only covering the re-run tests and not all the tests we did.

To fix this we used a workaround using the `--raw-command` flag to use our custom command on each rerun and write the coverage in a different file for each run. The custom script `go test` line looked like this (for linux but same idea for windows):
```
go test {gobuild_flags} {govet_flags} {gotest_flags} -json -coverprofile=\"$(mktemp {TMP_PROFILE_COV_PREFIX}.XXXXXXXXXX)\" $@
```

**TL; DR**: we were passing twice all the `go test` flags:
- `{gobuild_flags} {govet_flags} {gotest_flags}`
- `$@` which makes `gotestsum` pass all the `go test` flags passed in the first run + [2 added specific re-run flags](https://github.com/gotestyourself/gotestsum/blob/bc98120bd1b517a70af968159c46bb52a129429f/README.md?plain=1#L214-L217).

Among those 2 added flags was [the `-test.run` flag](https://github.com/gotestyourself/gotestsum/blob/bc98120bd1b517a70af968159c46bb52a129429f/README.md?plain=1#L215) (added by `gotestsum`) that filters the tests needed to be re-run.

Doubling the flags added a bunch of `--` making the `-test.run` flag get ignored, re-running instead all the tests. **This made `gotestsum` re-running all the tests for each failed test for each retry**, skyrocketing our jobs' runtime.

[It's now working as expected](https://github.com/DataDog/datadog-agent/actions/runs/8112901538/job/22175077214) !

### Additional Notes

Hopefully https://github.com/gotestyourself/gotestsum/pull/387 gets merge, allowing us to remove all this boilerplate ! 🤞

### CODE REVIEW

**Comment by amenasria** [comment]
Cleaning the code:

**Review by spencergilbert — APPROVED** [review]
